src/node/markdownToVue.ts

```diff
@@ -243,18 +244,21 @@ function genPageDataCode(tags: string[], data: PageData, replaceRegex: RegExp) {
   return tags
 }
 
-const inferTitle = (frontmatter: Record<string, any>, content: string) => {
-  if (frontmatter.title) {
-    return deeplyParseHeader(frontmatter.title)
-  }
-
-  const match = content.match(/^\s*#+\s+(.*)/m)
-
-  if (match) {
-    return deeplyParseHeader(match[1].trim())
+const inferTitle = (
+  md: MarkdownRenderer,
+  frontmatter: Record<string, any>,
+  title: string
+) => {
+  if (typeof frontmatter.title === 'string') {
+    const titleToken = md.parseInline(frontmatter.title, {})[0]
```

Comment:
Why are we parsing frontmatter.title? Do people actually put random markdown/html syntax there?

Reply:
It should be migrated from vuepress v1, or comes from some feature request?

As you can see, we were using `deeplyParseHeader(frontmatter.title)` before.

Reply:
Yes it's from vuepress v1: https://github.com/vuejs/vuepress/blob/d692cc852a26ed343f26a7bdd9d336ad3b0da191/packages/%40vuepress/shared-utils/src/inferTitle.ts#L11-L22

In fact, I didn't migrate it to vuepress v2, because I don't think people may expect markdown syntax to be worked in frontmatter (I think you might have similar thoughts lol). The only thing might be the emoji syntax `:foo:`

Reply:
Ah fine, let's keep it this way. 😅